src/chainlit/server.py

```diff
@@ -325,17 +325,13 @@ async def serve(path: str):
 # -------------------------------------------------------------------------------
 
 
-def need_session(id: str):
-    """Return the session with the given id."""
-
-    session = sessions.get(id)
-    if not session:
-        raise ValueError("Session not found")
-    return session
-
-
 @socket.on("connect")
-async def connect(sid, environ):
+async def connect(sid, environ, auth):
+    session_id = auth and auth.get("sessionId")
```

Comment:
so here if a session id leaked somehow I could restore a session from another user. how likely is it to happen?

Reply:
I think the attack surface is pretty low, owasp recommends at least 128 bits unpredictable session id, here we are using 256 bits.
In my opinion, the risk of hijacking a session is pretty the same that getting access to the auth token.
However checking the email can eventually mitigate the risk by the fact that an access token is theoretically short lived.

Do you want to add it in this PR or merge this part and do it in another?

https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#session-id-length

Reply:
we can do it in a different one